docs/branching_model_team_agreement.md

```diff
@@ -3,16 +3,18 @@
 This document lists the agreed upon practices the team will be using. As they are discussed and changed, update this file accordingly.
 
 ## Guardrails
-- The default branch for this repo is `main`
-- all commits must be made to a non-protected branch and submitted via a pull request before they can be merged into `main`. (There can be no direct commits made to `main`.)
-- Pull requests require at least 1 approval before the can be merged to `main`. Any commits made to an approved branch will dismiss the approval. (If you commit new code to a branch after it gets approved but before you merge it, you have to get it approved again.)
-- Pull requests must pass CI checks in order to be merged to `main`
-- Branches are automatically deleted when merged
-- Squash merge has been disabled
+The following guardrails are in place to ensure we keep a clean working tree, while avoiding merging breaking changes into our main branch.
+- The default branch for this repo is `main`.
+- All commits must be made to a non-protected branch and submitted via a pull request before they can be merged into `main`. (There can be no direct commits made to `main`.)
+- Pull requests require at least 1 approval before they can be merged to `main`. Any commits made to an approved branch will dismiss the approval. (If you commit new code to a branch after it gets approved but before you merge it, you have to get it approved again.)
+- Pull requests must pass CI checks in order to be merged to `main`.
+- Branches are automatically deleted on Github when a PR is merged. Each teammate should set `git config fetch.prune true` to mirror this behaviour to their local workstation.
```

Comment:
Having the `fetch.prune true` on our local machine, will that only delete the branch once it's merged into main on remote?

Reply:
`git fetch --prune is the best utility for cleaning outdated branches. It will connect to a shared remote repository remote and fetch all remote branch refs. It will then delete remote refs that are no longer in use on the remote repository.`
From: https://www.atlassian.com/git/tutorials/git-prune#:~:text=git%20fetch%20%2D%2Dprune%20is,use%20on%20the%20remote%20repository.

Essentially it keeps you more in sync with the branching on origin (especially with branches being deleted on merge), by running prune whenever you fetch or pull.